ladybug/datacollection.py

```diff
@@ -145,8 +149,77 @@ def data(self):
         return self._data
 
     def duplicate(self):
-        """Duplicate current data list."""
-        return DataCollection(self.data, self.header)
+        """Return a copy of the current data list."""
+        return DataCollection(self.data, self.header.duplicate())
+
+    def is_in_range(self, lower=float('-inf'), upper=float('+inf'),
+                    raise_exception=False):
+        """Check if the values are in within lower and upper limits."""
+        for value in self.values:
+            if value < lower or value > upper:
+                if not raise_exception:
+                    return False
+                else:
+                    raise ValueError(
+                        'Values should be between {1} and {2}. Got {3}'.format(
+                            lower, upper, value
+                        )
+                    )
+        return True
+
+    def is_in_range_data_type(self, raise_exception=False):
```

Comment:
I have the same question here. Why would one use this and when?

Reply:
I could imagine hiding this function since the purpose of it is to make sure that the values in the DataCollection are within the ranges that are physically or mathematically possible for the DataType.  For example, we usually want to make sure that the values in a Temperature DataCollection do not have values below absolute zero.  Or we want to make sure that there are no negative values in a solar radiation DataCollection.  Since this is something that will probably be run automatically when I push the next PR, I can hide it if you want.  Just let me know.

Reply:
If we hide this method then do we call this under __init__? How do you see this to be used in the process?

Reply:
Yes, I would see it being called during __init__().  I have hidden the function for now and I will add it to __init__() in the next PER that I send that will break up the DataCollection class.